Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -239,10 +246,15 @@ def bottle_formula(f)
           tar_path.utime(tab.source_modified_time, tab.source_modified_time)
           relocatable_tar_path = "#{f}-bottle.tar"
           mv tar_path, relocatable_tar_path
-          # Use gzip, faster to compress than bzip2, faster to uncompress than bzip2
-          # or an uncompressed tarball (and more bandwidth friendly).
-          safe_system "gzip", "-f", relocatable_tar_path
-          mv "#{relocatable_tar_path}.gz", bottle_path
+          case compression_type
+          when :xz
+            safe_system "#{HOMEBREW_PREFIX}/opt/xz/bin/xz", "-f", relocatable_tar_path
```

Comment:
I'd just use a blanket `xz` here and ensure it was added to the `PATH` previously if needed.